skops/io/_utils.py

```diff
@@ -171,3 +171,21 @@ def get_state(value, save_context):
 
     res["__id__"] = __id__
     return res
+
+
+def get_type_name(
+    t: Any,
+) -> str:
+    """Simple helper function to take in a type, and return its name as a string"""
+    return f"{t.__module__}.{t.__name__}"
+
+
+def cast_defaults_to_strings(defaults: List[Any]) -> List[str]:
+    """Simple helper function that takes in a list of defaults,
+    and converts any types found to strings"""
+    if not defaults:
+        return []
+    if not type(defaults) in [list, tuple]:
```

Comment:
Ah I see, I wasn't fully super familiar with mypy, that's handy! I'll rework this to accept `Any` instead of just `list`s

So in general, for internal functions, given we use mypy, are type checks like this best practice, or should we rely on mypy for that?

Reply:
Technically, mypy should warn if the wrong type is passed and thus it shouldn't happen. In practice, since not all our code is typed, or some functions just return `Any`, we could still end up calling this function with a non-list without mypy warning us. Therefore, I think the check has its merit and at the same time, we should keep the annotation, so scratch what I said earlier.

Usually, I would be fine not checking and failing when a wrong type is passed (i.e. rely on duck typing), the problem is that if a user were to pass a string, since it's iterable, it would work but the totally wrong thing would happen, so the check is needed.

Just note that this time, I think it's really better to check `isinstance(defaults, (list, tuple))` :)

Reply:
I wouldn't rely on mypy for runtime type checks, those are two different things. The functions should still check for validity of their input.

Reply:
Well, the majority of our functions don't validate the input, right? :) So I think it's more of a matter of checking where there is a reasonable error case that would be missed otherwise.

As to mypy, if we had a strictly typed code base (which we chose not to enforce), all internal functions would only be called by the annotated types, otherwise mypy would complain (excluding mypy bugs).